src/jit/morph.cpp

```diff
@@ -2953,23 +2951,19 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* callNode)
         // this can't be a struct.
         assert(argx->gtType != TYP_STRUCT);
 
-        // FIXME: Issue #4025 Why do we need floating type for 'this' argument
```

Comment:
You deleted this comment, but it's still not clear why we need to check for floating point on 'this'.